diff --git a/meson.build b/meson.build
index b6939c7..a10b725 100644
--- a/meson.build
+++ b/meson.build
@@ -710,42 +710,44 @@ executable(
 
 ## testdso module
 
-testdso_mod = shared_module(
-    'testdso',
-    files('testdso.c'),
-    build_rpath: get_option('libdir'),
-    include_directories: config_dir,
-    name_prefix: '',
-)
-
 ## tests
 
-checks = [
-    'runsuite',
-    'runtest',
-    'runxmlconf',
-# Disabled for now, see #694
-#    'testModule',
-    'testThreads',
-    'testapi',
-    'testchar',
-    'testdict',
-    'testlimits',
-    'testparser',
-    'testrecurse',
-]
-
-foreach check : checks
-    exe = executable(
-        check,
-        files(check + '.c'),
-        dependencies: [threads_dep, xml_dep],
-        include_directories: config_dir,
-    )
-    if check != 'testlimits'
-        test(check, exe, timeout: 0, workdir: meson.current_source_dir())
-    endif
-endforeach
+if not get_option('tests').disabled()
+  testdso_mod = shared_module(
+      'testdso',
+      files('testdso.c'),
+      build_rpath: get_option('libdir'),
+      include_directories: config_dir,
+      name_prefix: '',
+  )
+
+  checks = [
+      'runsuite',
+      'runtest',
+      'runxmlconf',
+  # Disabled for now, see #694
+  #    'testModule',
+      'testThreads',
+      'testapi',
+      'testchar',
+      'testdict',
+      'testlimits',
+      'testparser',
+      'testrecurse',
+  ]
+
+  foreach check : checks
+      exe = executable(
+          check,
+          files(check + '.c'),
+          dependencies: [threads_dep, xml_dep],
+          include_directories: config_dir,
+      )
+      if check != 'testlimits'
+          test(check, exe, timeout: 0, workdir: meson.current_source_dir())
+      endif
+  endforeach
+endif
 
 subdir('example')
 subdir('doc')
diff --git a/meson_options.txt b/meson_options.txt
index 4d61f3d..dc67d17 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -251,3 +251,9 @@ option('minimum',
   value: false,
   description: 'build a minimally sized library (default=false)'
 )
+
+option('tests',
+  type: 'feature',
+  description: 'Build and run tests',
+  yield: true
+)
